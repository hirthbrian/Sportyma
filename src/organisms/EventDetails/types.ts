export interface Props {
  title: string;
  startDate: Date;
  endDate: Date;
  location: {
    place: string;
    address: string;
    city: string;
  };
}
