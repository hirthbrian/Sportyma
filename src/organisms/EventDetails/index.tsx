import { Platform, Linking } from "react-native";
import { format } from "date-fns";

import SectionTitle from "../../atoms/SectionTitle";

import DetailContainer from "../../molecules/DetailContainer";

import { Props } from "./types";

const calendarIcon = require("../../../assets/calendar.png");
const positionIcon = require("../../../assets/location.png");

const capitalizeFirstLetter = (text: string) =>
  text.charAt(0).toUpperCase() + text.slice(1);

const EventDetails = ({ title, startDate, endDate, location }: Props) => {
  const fullDate = capitalizeFirstLetter(
    format(new Date(startDate), "eeee d MMMM yyyy")
  );
  const dateHours = `${format(new Date(startDate), "HH:mm")} - ${format(
    new Date(endDate),
    "HH:mm"
  )}`;

  const onLocationPress = () => {
    const fullAddress = `${location.place} ${location.address} ${location.city}`;
    const url = Platform.select({
      ios: `maps:0,0?q=${fullAddress}`,
      android: `geo:0,0?q=${fullAddress}`,
    });

    Linking.openURL(url);
  };

  return (
    <>
      <SectionTitle title={title} />
      <DetailContainer
        title={fullDate}
        subTitle={dateHours}
        icon={calendarIcon}
      />
      <DetailContainer
        title={location.place}
        subTitle={`${location.address},\n${location.city}`}
        icon={positionIcon}
        onPress={onLocationPress}
      />
    </>
  );
};

export default EventDetails;
