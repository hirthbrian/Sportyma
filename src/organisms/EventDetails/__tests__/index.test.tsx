import React from "react";
import renderer from "react-test-renderer";

import EventDetails from "..";

const startDate = "2023-02-16T20:35:00+00:00" as unknown as Date;
const endDate = "2023-02-16T20:35:00+00:00" as unknown as Date;

describe("<EventDetails />", () => {
  let props;

  beforeEach(() => {
    props = {
      title: "test",
      startDate,
      endDate,
      location: {
        place: "Stade de la Meinau",
        address: "12 rue de lâ€™Extenwoerth",
        city: "67100 Strasbourg",
      },
    };
  });

  it("renders correcly", () => {
    expect(renderer.create(<EventDetails {...props} />).toJSON()).toBeTruthy();
  });
  it("match snapshot", () => {
    const tree = renderer.create(<EventDetails {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
