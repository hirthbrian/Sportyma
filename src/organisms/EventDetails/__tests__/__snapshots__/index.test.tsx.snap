// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EventDetails /> match snapshot 1`] = `
Array [
  <Text
    size={20}
    style={
      Array [
        Object {
          "color": "#152D4B",
          "fontFamily": "Kanit-Medium",
          "fontSize": 20,
        },
      ]
    }
    weight={500}
  >
    test
  </Text>,
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "flexDirection": "row",
          "paddingTop": 15,
        },
      ]
    }
  >
    <View
      color="#E3E8F9"
      size={36}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#E3E8F9",
            "borderRadius": 18,
            "height": 36,
            "justifyContent": "center",
            "marginRight": 10,
            "width": 36,
          },
        ]
      }
    >
      <Image
        color="#2559F2"
        source={1}
        style={
          Array [
            Object {
              "height": 18,
              "tintColor": "#2559F2",
              "width": 18,
            },
          ]
        }
      />
    </View>
    <View>
      <Text
        style={
          Array [
            Object {
              "color": "#152D4B",
              "fontFamily": "Kanit-Regular",
              "fontSize": 16,
            },
          ]
        }
      >
        Thursday 16 February 2023
      </Text>
      <Text
        color="#778EB9"
        style={
          Array [
            Object {
              "color": "#778EB9",
              "fontFamily": "Kanit-Light",
              "fontSize": 16,
            },
          ]
        }
        weight={300}
      >
        21:35 - 21:35
      </Text>
    </View>
  </View>,
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "flexDirection": "row",
          "paddingTop": 15,
        },
      ]
    }
  >
    <View
      color="#E3E8F9"
      size={36}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#E3E8F9",
            "borderRadius": 18,
            "height": 36,
            "justifyContent": "center",
            "marginRight": 10,
            "width": 36,
          },
        ]
      }
    >
      <Image
        color="#2559F2"
        source={1}
        style={
          Array [
            Object {
              "height": 18,
              "tintColor": "#2559F2",
              "width": 18,
            },
          ]
        }
      />
    </View>
    <View>
      <Text
        style={
          Array [
            Object {
              "color": "#152D4B",
              "fontFamily": "Kanit-Regular",
              "fontSize": 16,
            },
          ]
        }
      >
        Stade de la Meinau
      </Text>
      <Text
        color="#778EB9"
        style={
          Array [
            Object {
              "color": "#778EB9",
              "fontFamily": "Kanit-Light",
              "fontSize": 16,
            },
          ]
        }
        weight={300}
      >
        12 rue de lâ€™Extenwoerth,
67100 Strasbourg
      </Text>
    </View>
  </View>,
]
`;
